src/neo4j_genai/types.py

```diff
@@ -27,17 +27,57 @@
 from neo4j_genai.retrievers.utils import validate_search_query_input
 
 
-class VectorSearchRecord(BaseModel):
+class RetrieverResultItem(BaseModel):
     """
-    Represents a record returned from a vector search.
+    A single record returned from a retriever.
 
     Attributes:
-        node (Any): The node data retrieved by the search.
-        score (float): The similarity score of the retrieved embedding.
+        content (str): The context as will be provided to the LLM
+        metadata (Optional[dict]): Any metadata that can be included together
+            with the text, related to that record (e.g. another node property)
     """
 
-    node: Any
-    score: float
+    content: str
+    metadata: Optional[dict] = None
+
+
+class RetrieverRawResult(BaseModel):
+    """
+    Represents the raw result returned from the retriever get_search_result
+    method. It needs to be formatted further before being returned as a RetrieverResult.
+
+    Attributes:
+        records (list[neo4j.Record]): A list of records from neo4j.
+        metadata: Record-related metadata, such as score.
+    """
+
+    records: list[neo4j.Record]
```

Comment:
Is it possible to merge RetrieverRawResult and RetrieverResult, such that RetrieverResultItem either contains str_content or record_content? 
As the result user wouldn't need to think/learn when to use either of them

Reply:
The `get_search_results` method returns `neo4j.Record`, but the results we need for RAG is a string. That why we introduced both models. It also makes sense to have a string returned here, since the way to extract the text content depends on the retriever.

Reply:
`RetrieverRawResult` does make me think that it's a parent of `RetrieverResult` though. Perhaps we can rename `RetrieverRawResult` to something like `RawSearchResult`? As it's a result for the retriever directly

Reply:
Thinking again about that : 
- The inconvenient is that users who may want to test different formatting will have to re-run the retriever.
- Advantage is that when we will have agents, we will be able to have one formatting per retriever for free, if we postpone the formatting it will be harder to have something other than `str(retriever_result)`.